src/file_handle_cache.erl

```diff
@@ -581,6 +583,46 @@ info_keys() -> ?INFO_KEYS.
 info() -> info(?INFO_KEYS).
 info(Items) -> gen_server2:call(?SERVER, {info, Items}, infinity).
 
+clear_read_cache() ->
+    gen_server2:cast(?SERVER, clear_read_cache),
+    clear_vhost_read_cache(rabbit_vhost:list()).
+
+clear_vhost_read_cache([]) ->
+    ok;
+clear_vhost_read_cache([VHost | Rest]) ->
+    clear_queue_read_cache(rabbit_amqqueue:list(VHost)),
+    clear_vhost_read_cache(Rest).
+
+clear_queue_read_cache([]) ->
+    ok;
+clear_queue_read_cache(
+  [#amqqueue{name = #resource{virtual_host = VHost, name = Name},
+             pid = MPid, slave_pids = SPids}
+  | Rest]) ->
+    rabbit_log:info(
+      "Clear read buffer cache for queue \"~s\" on vhost \"~s\"~n",
+      [Name, VHost]),
+    Fun = fun(_, State) ->
+        clear_process_read_cache(),
+        State
+    end,
+    rabbit_amqqueue:run_backing_queue(MPid, rabbit_variable_queue, Fun),
+    [rabbit_amqqueue:run_backing_queue(SPid, rabbit_variable_queue, Fun)
```

Comment:
Do we need to clear the cache on the slave Pids? Wouldn't that clear caches on a separate node?